web/api/v1/api.go

```diff
@@ -703,6 +708,12 @@ func (api *API) labelNames(r *http.Request) apiFuncResult {
 	if names == nil {
 		names = []string{}
 	}
+
+	if len(names) >= limit {
+		names = names[:limit]
+		warnings := annotations.New().Add(errors.New("results truncated due to limit"))
```

Comment:
```suggestion
		warnings = warnings.Add(errors.New("results truncated due to limit"))
```

`:=` is a mistake: it will ignore any previous warnings and create a new local variable.